### PR TITLE
WIP add timeout option

### PR DESCRIPTION
**Pull request by aoberoi**
starts to address #56. the implementation probably works, but the testing is blocked on gabrielfalcao/HTTPretty#202

### CODE REVIEW

**Comment by aoberoi** [comment]
do not merge yet.

**Comment by coulix** [comment]
Can we merge this one?

**Comment by aoberoi** [comment]
@coulix: no, not yet. The blocking issue hasn't been closed. If you could :+1: the PR linked above, maybe that will help get it released.

**Comment by coulix** [comment]
:+1:

File: tests/test_http_options.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    def tearDown(self):
+        httpretty.disable()
+
+    def test_timeout(self):
+        # TODO: tell httpretty to timeout
+        opentok = OpenTok(self.api_key, self.api_secret, timeout=5)
+
+        session = opentok.create_session()
+
```
Comment (coulix) [inline tests/test_http_options.py]: To test the exception we could do something like this.

    # create a callback body that raises an exception when opened
    def exceptionCallback(request, uri, headers):

        # raise your favourite exception here, e.g. requests.ConnectionError or requests.Timeout
        raise requests.Timeout('Connection timed out.')

    # set up a mock and use the callback function as response's body
    httpretty.register_uri(
            method=httpretty.GET,
            uri='https://api.opentok.com/session/create',
            status=200,
            body=exceptionCallback
        )

    with self.assertRaises(requests.Timeout):
        opentok = OpenTok(self.api_key, self.api_secret, timeout=5
        session = opentok.create_session()

**Comment by tylergould** [comment]
@aoberoi it looks like the fix for the testing issue has been released, if you're still interested in merging this one

**Pull request merged**